lib/extract-templates.js

```diff
@@ -0,0 +1,133 @@
+import { parseTemplates } from 'ember-template-imports/lib/parse-templates.js';
+
+export const SUPPORTED_EXTENSIONS = ['.js', '.ts', '.gjs', '.gts'];
+const LOCATION_START = Object.freeze({ line: 0, column: 0 });
+/**
+ * @typedef {object} TemplateInfo
+ * @property {number} line
+ * @property {number} column
+ * @property {string} template
+ * @property {boolean} isEmbedded
+ *
+ * @param {string} moduleSource
+ * @param {string} relativePath
+ *
+ * @returns {TemplateInfo[]}
+ */
+export function extractTemplates(moduleSource, relativePath) {
+  // If no relativePath is present, assuming we might have templates.
+  let mightHaveTemplates = relativePath
+    ? SUPPORTED_EXTENSIONS.some((ext) => relativePath.endsWith(ext))
+    : true;
+
+  if (!mightHaveTemplates) {
+    return [makeTemplateInfo(LOCATION_START, moduleSource)];
+  }
+
+  let parsed = parseTemplates(moduleSource, relativePath, 'template');
+
+  // If after parsing we found no templates and we had no relative path,
+  // then we assume we had a hbs file in input.
+  if (parsed.length === 0 && !relativePath) {
+    return [makeTemplateInfo(LOCATION_START, moduleSource)];
+  }
+
+  let result = parsed.map((templateInfo) => {
+    let { start, end } = templateInfo;
+    let templateStart = start.index + start[0].length;
+
+    let template = moduleSource.slice(templateStart, end.index);
+
+    return makeTemplateInfo(coordinatesOf(moduleSource, templateStart, end.index), template, true);
+  });
+
+  return result;
+}
+
+/**
+ * @param {object} location
+ * @param {number} location.line
+ * @param {number} location.column
+ * @param {string} template
+ * @param {boolean} isEmbedded
+ *
+ * @returns {TemplateInfo}
+ */
+function makeTemplateInfo({ line, column }, template, isEmbedded) {
+  return {
+    line,
+    column,
+    template,
+    isEmbedded,
+  };
+}
+
+export function coordinatesOf(text, offset) {
+  const contentBeforeTemplateStart = text.slice(0, offset).split('\n');
+  return {
+    line: contentBeforeTemplateStart.length,
+    column: contentBeforeTemplateStart[contentBeforeTemplateStart.length - 1].length,
+  };
+}
+
+/**
+ * @param {TemplateInfo} templateInfo
+ * @param {object} result
+ * @param {number} result.line
+ * @param {number} result.endLine
+ * @param {number} result.column
+ * @param {number} result.endColumn
+ *
+ * @returns {object}
+ */
+export function coordinatesOfResult(templateInfo, result) {
+  /**
```

Comment:
Can you update the docs here to have a high level explanation of what the point of the function is? I assume it's something like

```suggestion
/**
 * Processes results and corrects for template location offsets.
 *
 * @param {TemplateInfo} templateInfo
 * @param {object} result
 * @param {number} result.line
 * @param {number} result.endLine
 * @param {number} result.column
 * @param {number} result.endColumn
 *
 * @returns {object}
 */
export function coordinatesOfResult(templateInfo, result) {
  /**
```

Reply:
I think this is still outstanding (need a description added to the docs)

Reply:
Sorry about that. I must have missed this somehow.